modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/EntityModelSchemaBasedEdmProvider.java

```diff
@@ -325,6 +331,33 @@ private FullQualifiedName _getFullQualifiedName(EntityModel entityModel) {
 		return new FullQualifiedName(_NAMESPACE + "." + entityModel.getName());
 	}
 
+	private List<CsdlComplexType> _handleComplexTypes(
+		List<CsdlComplexType> csdlComplexTypes) {
+
+		Map<String, CsdlComplexType> csdlComplexTypeMap = new HashMap<>();
+
+		for (CsdlComplexType csdlComplexType : csdlComplexTypes) {
+			CsdlComplexType defaultCsdlComplexType =
+				csdlComplexTypeMap.getOrDefault(
+					csdlComplexType.getName(), csdlComplexType);
+
+			List<CsdlProperty> defaultCsdlComplexTypeProperties =
+				defaultCsdlComplexType.getProperties();
+
+			List<CsdlProperty> csdlComplexTypeProperties =
+				csdlComplexType.getProperties();
+
+			if (defaultCsdlComplexTypeProperties.size() <=
```

Comment:
As we discussed, with the lower depth I think we could fix this problem in a better way

Reply:
Done